### PR TITLE
no_std support:

### PR DESCRIPTION
**Pull request by MabezDev**
* Adds the std feature which is enabled by default
* Uses libcore where possible to support no_std targets

### CODE REVIEW

**Comment by IvanUkhov** [comment]
Thank you! Are there any crates doing it like this?

**Comment by MabezDev** [comment]
Almost all optionally no_std crates do this in some form as far as I'm aware. [smoltcp](https://github.com/m-labs/smoltcp), [fatfs](https://github.com/rafalh/rust-fatfs), [byteorder](https://github.com/BurntSushi/byteorder) just to name a few.

File: src/lib.rs [hunk 1]
```diff
@@ -28,10 +28,16 @@
 // https://people.csail.mit.edu/rivest/Md5.c
 // https://tools.ietf.org/html/rfc1321
 
-use std::convert;
-use std::fmt;
-use std::io;
-use std::ops;
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
```
Comment (IvanUkhov) [inline src/lib.rs]: Why do you want to have `test` here? Probably one should change [.travis.yaml](https://github.com/stainless-steel/md5/blob/master/.travis.yml#L10) to run everything with and without the `std` feature. What do you think?

File: src/lib.rs [hunk 2]
```diff
@@ -28,10 +28,16 @@
...
+
+#[cfg(feature = "std")]
+use std as core;
+
+use core::convert;
+use core::fmt;
+use core::ops;
+#[cfg(feature = "std")]
```
Comment (IvanUkhov) [inline src/lib.rs:40]: Let’s add a new line above to separate feature-dependent imports.

File: src/lib.rs (hunk 1 above)
Comment (MabezDev) [reply src/lib.rs]: I guess test isn't technically needed. Testing the two seperately doesn't make much sense as testing the std feature, already tests all the no_std code (cargo features are additive).

These changes effectively turn the crate into a no_std crate, that by default, opts into std features, does that make more sense?

Comment (IvanUkhov) [reply src/lib.rs]: I mean, some `#![…]` might be misplaced, and it won’t work once the feature is disabled. It’s not a check for the written tests, but for being able to compile without the feature.

Comment (MabezDev) [reply src/lib.rs]: That makes sense, I've added a no_std build to travis.

**Comment by IvanUkhov** [comment]
Thank you!

**Pull request merged**